### PR TITLE
[TechDocs] clean up techdocs backend api report

### PR DESCRIPTION
**Pull request by emmaindal**
#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [ ] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/techdocs-backend/src/search/DefaultTechDocsCollator.ts [hunk 1]
```diff
@@ -55,46 +60,43 @@ type EntityInfo = {
...
-  private readonly tokenManager: TokenManager;
-  private readonly parallelismLimit: number;
-  private readonly legacyPathCasing: boolean;
   public readonly type: string = 'techdocs';
   public readonly visibilityPermission = catalogEntityReadPermission;
 
-  /**
-   * @deprecated use static fromConfig method instead.
-   */
-  constructor(options: TechDocsCollatorOptions) {
```
Comment (iamEAP) [inline plugins/techdocs-backend/src/search/DefaultTechDocsCollator.ts:72]: A `minor` changeset for this change, eh?  With instructions on how to update existing backend code.

File: plugins/techdocs-backend/src/service/router.ts [hunk 2]
```diff
@@ -323,12 +336,8 @@ export function createEventStream(
 }
 
 /**
- * Create a HTTP response. This is used for the legacy non-event-stream implementation of the sync endpoint.
- *
- * @param res - the response to write the event-stream to
- * @returns A tuple of <log, error, finish> callbacks to emit messages. A call to 'error' or 'finish'
- *          will close the event-stream.
- */
+ *  @deprecated use event-stream implementation of the sync endpoint
```
Comment (iamEAP) [inline plugins/techdocs-backend/src/service/router.ts:339]: Kind of a bummer that this doesn't get captured in the `api-report.md`, eh?  Maybe worth noting in a changeset too.  Something like:

> _Note: in an upcoming release, TechDocs backend's `/sync/:namespace/:kind/:name` endpoint will only respond to `text/event-stream`-based requests. Update any custom code at your organization accordingly._

File: plugins/techdocs-backend/src/search/DefaultTechDocsCollator.ts [hunk 3]
```diff
@@ -55,46 +60,43 @@ type EntityInfo = {
   kind: string;
 };
 
+/**
+ * A search collator responsible for gather and transform TechDocs document to index.
```
Comment (iamEAP) [inline plugins/techdocs-backend/src/search/DefaultTechDocsCollator.ts:64]:
```suggestion
 * A search collator responsible for gathering and transforming TechDocs documents.
```

Maybe?

**Review by iamEAP — COMMENTED** [review]
Changes look good!  I think this just needs a changeset.

File: plugins/techdocs-backend/src/service/router.ts (hunk 2 above)
Comment (emmaindal) [reply plugins/techdocs-backend/src/service/router.ts:339]: Yeah agree, but why does it not get captured in the api report? I decided to add the deprecated flag anyways. Will add instructions to changeset!

**Review by iamEAP — APPROVED** [review]

**Review by freben — APPROVED** [review]

**Pull request merged**